src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-builder.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+
+import { ServicesModule } from '../services/services.module';
+import { CdFormGroup } from './cd-form-group';
+
+/**
+ * CdFormBuilder extends FormBuilder to create an CdFormGroup based form.
+ */
+@Injectable({
+  providedIn: ServicesModule
+})
+export class CdFormBuilder extends FormBuilder {
```

Comment:
This could be simplified, by doing the following:
```Diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-builder.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-builder.ts
index 465e9763d1..8ce3be7854 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-builder.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-form-builder.ts
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
+import { FormBuilder } from '@angular/forms';

 import { ServicesModule } from '../services/services.module';
 import { CdFormGroup } from './cd-form-group';
@@ -11,34 +11,8 @@ import { CdFormGroup } from './cd-form-group';
   providedIn: ServicesModule
 })
 export class CdFormBuilder extends FormBuilder {
-  group(
-    controlsConfig: { [key: string]: any },
-    extra: { [key: string]: any } | null = null
-  ): CdFormGroup {
-    return this._convertFormGroupToCdFormGroup(super.group(controlsConfig, extra));
-  }
-
-  /**
-   * It actually converts a FormGroup to a CdFormGroup.
-   * This way the 'group' method of FormBuilder has not to be reworked, this means it will still,
-   * work as expected as long as the parameters of 'group' are not changed.
-   *
-   * On the downside this has to be updated when CdFormGroup has new methods or
-   * changed method names.
-   *
-   * @param {FormGroup} formGroup
-   * @returns {CdFormGroup}
-   * @private
-   */
-  _convertFormGroupToCdFormGroup(formGroup: FormGroup): CdFormGroup {
-    const form = formGroup as CdFormGroup;
-    form.constructor = CdFormGroup.prototype.constructor;
-    form.get = CdFormGroup.prototype.get;
-    form._get = CdFormGroup.prototype._get;
-    form.getValue = CdFormGroup.prototype.getValue;
-    form.silentSet = CdFormGroup.prototype.silentSet;
-    form._filterValue = CdFormGroup.prototype._filterValue;
-    form['__proto__'] = CdFormGroup.prototype;
-    return form;
+  group(controlsConfig: object, extra: object | null = null): CdFormGroup {
+    const formGroup = super.group(controlsConfig, extra);
+    return new CdFormGroup(formGroup.controls, formGroup.validator, formGroup.asyncValidator);
   }
 }
```